source/Calamari.Tests/KubernetesFixtures/ManifestReporterTests.cs

```diff
@@ -0,0 +1,113 @@
+            }
+        }
+        
+        [Test]
+        public void GivenInValidManifest_ShouldNotPostServiceMessage()
+        {
+            var memoryLog = new InMemoryLog();
+            var variables = new CalamariVariables();
+            
+            var yaml = @"text - Bar";
+            using (CreteFile(yaml, out var filePath))
+            {
+                var mr = new ManifestReporter(variables, CalamariPhysicalFileSystem.GetPhysicalFileSystem(), memoryLog);
+
+                mr.ReportManifestApplied(filePath);
+                
+                memoryLog.ServiceMessages.Should().BeEmpty();
+            }
+        }
+        
+        [Test]
+        public void GivenNamespaceInManifest_ShouldReportManifestNamespace()
+        {
+            var memoryLog = new InMemoryLog();
+            var variables = new CalamariVariables();
+            var yaml = @"metadata:
+  name: game-demo
+  namespace: XXX";
+            using (CreteFile(yaml, out var filePath))
+            {
+                var variableNs = Some.String();
+                variables.Set(SpecialVariables.Namespace, variableNs);
+                var mr = new ManifestReporter(variables, CalamariPhysicalFileSystem.GetPhysicalFileSystem(), memoryLog);
+
+                mr.ReportManifestApplied(filePath);
+
+                memoryLog.ServiceMessages.First().Properties.Should().Contain(new KeyValuePair<string, string>("ns", "XXX")); 
+            }
+        }
+        
+        [Test]
+        public void GivenNamespaceNotInManifest_ShouldReportVariableNamespace()
+        {
+            var memoryLog = new InMemoryLog();
+            var variables = new CalamariVariables();
+            var yaml = @"foo: bar";
+            using (CreteFile(yaml, out var filePath))
+            {
+                var variableNs = Some.String();
+                variables.Set(SpecialVariables.Namespace, variableNs);
+                var mr = new ManifestReporter(variables, CalamariPhysicalFileSystem.GetPhysicalFileSystem(), memoryLog);
+
+                mr.ReportManifestApplied(filePath);
+
+                memoryLog.ServiceMessages.First().Properties.Should().Contain(new KeyValuePair<string, string>("ns", variableNs)); 
+            }
+        }
+        
+        [Test]
+        public void GiveNoNamespaces_ShouldDefaultNamespace()
+        {
+            var memoryLog = new InMemoryLog();
+            var variables = new CalamariVariables();
+            var yaml = @"foo: bar";
+            using (CreteFile(yaml, out var filePath))
+            {
+                var mr = new ManifestReporter(variables, CalamariPhysicalFileSystem.GetPhysicalFileSystem(), memoryLog);
+
+                mr.ReportManifestApplied(filePath);
+
+                memoryLog.ServiceMessages.First().Properties.Should().Contain(new KeyValuePair<string, string>("ns", "default")); 
+            }
+        }
+
+        IDisposable CreteFile(string yaml, out string filePath)
```

Comment:
I thought this was a fancy library named after the island :greece:  before I saw it was a local method :joy: